src/cache.rs

```diff
@@ -782,18 +808,37 @@ impl Cache {
                 cache_err
                     .unwrap_error("cache::prepare_stdlib(): expected standard library to be parsed")
             })?;
-        Ok(())
+        let eval_env = self.mk_global_eval().unwrap();
+        Ok(GlobalEnv { eval_env, type_env })
     }
 
-    /// Generate a global environment from the list of `file_ids` corresponding to the standard
+    /// Generate a global typing environment from the list of `file_ids` corresponding to the standard
     /// library parts.
-    pub fn mk_global_env(&self) -> Result<eval::Environment, CacheError<Void>> {
-        if let Some(ids) = self.stdlib_ids.as_ref().cloned() {
-            let mut env = eval::Environment::new();
+    pub fn mk_global_types(&self) -> Result<typecheck::Environment, CacheError<Void>> {
```

Comment:
As a nitpick, I would suggest to add `env` in the name of those functions, because I find we quite lose the meaning going from `mk_global_env` to `mk_global_eval` and `mk_global_types` (We don't really make an "eval"). Something like `mk_global_eval_env` and `mk_global_type_env` (or even dropping the `global`, if you find it too verbose, as it is a less useful information I think) may be more suggestive.

Reply:
Lol, I was thinking the oposit ;) both function return an `Environment` so the env part was useless for me. but you're right, it's not that evident actualy.